ontoma/owl.py

```diff
@@ -0,0 +1,71 @@
+import logging
+import os
+import urllib.request
+
+import pandas as pd
+import pronto
+import requests
+
+from ontoma import ontology
+
+EFO_RELEASE_API = 'https://api.github.com/repos/EBISPOT/efo/releases/latest'
+
+OWL_FILENAME = 'efo_otar_slim.owl'
+TERMS_FILENAME = 'terms.tsv'
+XREFS_FILENAME = 'xrefs.tsv'
+SYNONYMS_FILENAME = 'synonyms.tsv'
+
+ALL_FILENAMES = {OWL_FILENAME, TERMS_FILENAME, XREFS_FILENAME, SYNONYMS_FILENAME}
+
+logger = logging.getLogger(__name__)
+
+
+def preprocess_owl(outdir):
+    """Downloads and preprocesses the EFO OT slim CWL file. Outputs several tables in TSV format to the specified
+    directory."""
+
+    if not os.path.isdir(outdir):
+        logging.info('Create the output directory.')
+        os.mkdir(outdir)
+
+    if not os.path.isfile(os.path.join(outdir, OWL_FILENAME)):
+        logging.info('Download the EFO OT slim OWL file.')
+        efo_assets = requests.get(EFO_RELEASE_API).json()['assets']
```

Comment:
Absolutely, thank you for pointing this out. In the process of testing this I also discovered that the latest EFO release on GitHub does not contain the Open Targets slim file (temporarily, they say they'll fix it shortly). So the commit https://github.com/opentargets/OnToma/pull/18/commits/2a14e869c872dfec65f15e1282d1aa8b19e58720 does a number of things:
* Adds a switch to specify the EFO version to use (still latest by default)
* More informative error messages in case the release is incomplete
* Retry requests to the GitHub API in case they fail

Reply:
Great, this is fantastic!